subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/BasicGroupedTaskLoggingFunctionalSpec.groovy

```diff
@@ -206,6 +206,37 @@ class BasicGroupedTaskLoggingFunctionalSpec extends AbstractConsoleFunctionalSpe
         gradle?.waitForFinish()
     }
 
+    def "group header is printed red if task failed"() {
+        given:
+        buildFile << """
+            task failing { doFirst { 
+                logger.quiet 'hello'
+                throw new RuntimeException('Failure...')
+            } }
+        """
+
+        when:
+        fails('failing')
+
+        then:
+        result.output.contains("[31;1m> Task :failing")
+    }
+
+    def "group header is printed white if task succeeds"() {
+        given:
+        buildFile << """
+            task succeeding { doFirst { 
+                logger.quiet 'hello'
+            } }
+        """
+
+        when:
+        succeeds('succeeding')
+
+        then:
+        result.output.contains("[1m> Task :succeeding")
```

Comment:
Recommend use of `AbstractConsoleFunctionalSpec` instead of scraping output manually.